ext/storage_sources/s3_store/s3_connection.cpp

```diff
@@ -1,57 +1,78 @@
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include "s3_connection.h"
+
 #include <aws/s3-crt/model/DeleteObjectRequest.h>
 #include <aws/s3-crt/model/ListObjectsV2Request.h>
 #include <aws/s3-crt/model/PutObjectRequest.h>
 #include <aws/s3-crt/model/GetObjectRequest.h>
 #include <aws/s3-crt/model/HeadObjectRequest.h>
 #include <aws/s3-crt/model/HeadBucketRequest.h>
 
-#include "s3_connection.h"
-
 #include <fstream>
 #include <iostream>
-#include <string>
-#include <vector>
 
+// This is a tag that can be set and used when uploading or retrieving objects from the S3.
+// Tagging in S3 allows for categorisation of objects, as well as other benefits.
 #define S3_ALLOCATION_TAG ""
-/*
- * S3Connection --
- *     Constructor for AWS S3 bucket connection with provided credentials.
- */
+
+// Constructor for AWS S3 bucket conqnection with provided credentials.
 S3Connection::S3Connection(const Aws::Auth::AWSCredentials &credentials,
   const Aws::S3Crt::ClientConfiguration &config, const std::string &bucketName,
   const std::string &objPrefix)
     : _s3CrtClient(credentials, config), _bucketName(bucketName), _objectPrefix(objPrefix)
 {
-    /* Confirm that we can access the bucket, else fail. */
+    // Confirm that we can access the bucket, else fail.
     bool exists;
     int ret = BucketExists(exists);
-    if (ret != 0 || !exists)
+    if (!exists)
         throw std::invalid_argument(_bucketName + " : No such bucket.");
+    if (ret != 0)
+        throw std::invalid_argument(_bucketName + " :Unable to access bucket.");
 }
 
-/*
- * S3Connection --
- *     Constructor for AWS S3 bucket connection with credentials in local file.
- *     https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html
- */
+// Constructor for AWS S3 bucket connection with credentials in local file.
```

Comment:
I thought we decided to move all the comments to the header file.

Reply:
We had a discussion and decided to keep the function descriptions in the .cpp file and have a class description in the header file.